pkg/api/types.go

```diff
@@ -584,10 +584,36 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
+type HostPathType string
+
+const (
+	// If nothing exists at the given path, an empty directory will be created there.
+	HostPathUnset HostPathType = "DirectoryOrCreate"
+	// A directory entry of any type must exist at the given path
+	HostPathExists HostPathType = "Exists"
+	// A file must exist at the given path
+	HostPathFile HostPathType = "File"
+	// A block or character device must exist at the given path
+	HostPathDevice HostPathType = "Device"
+	// A socket must exist at the given path
+	HostPathSocket HostPathType = "Socket"
+	// A directory must exist at the given path
+	HostPathDirectory HostPathType = "Directory"
+	HostPathNewDir    HostPathType = "NewDirectory"
+	// Granularity beyond HostPathDevice shouldn't matter often
+	HostPathCharDev HostPathType = "CharDevice"
+	// Granularity beyond HostPathDevice shouldn't matter often
+	HostPathBlockDev HostPathType = "BlockDevice"
+	HostPathNewFile  HostPathType = "NewFile"
+	// If a path does not exist, then do not create any container-mount at all
+	HostPathOptional HostPathType = "Optional"
```

Comment:
This one is wildly different, and I'd rather add it in a subsequent PR, if we have real use cases.